### PR TITLE
Data retrieved using retrofit and okhttp

### PR DESCRIPTION
**Pull request by mananwason**
@creativepsyco Please review and feel free to merge.

### CODE REVIEW

File: app/src/androidTest/java/org/fossasia/openevent/APITest.java [hunk 1]
```diff
@@ -36,4 +38,25 @@ public void failure(RetrofitError error) {
         });
         latch.await();
     }
+
```
Comment (leto) [inline app/src/androidTest/java/org/fossasia/openevent/APITest.java]: Any reason this is commented out? Does it pass?

File: app/src/main/java/org/fossasia/openevent/activities/MainActivity.java [hunk 2]
```diff
@@ -84,7 +85,7 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
         setSupportActionBar((Toolbar) findViewById(R.id.toolbar));
-
+        //APIClient apiClient = new APIClient();
```
Comment (leto) [inline app/src/main/java/org/fossasia/openevent/activities/MainActivity.java]: Please remove dead code or explain what is going on here

File: app/src/main/java/org/fossasia/openevent/dbutils/DbSingleton.java [hunk 3]
```diff
@@ -108,23 +108,23 @@ public Session getSessionById (int id) {
...
-                    cur.getString(cur.getColumnIndex(DbContract.Speakers.FACEBOOK)),
-                    cur.getString(cur.getColumnIndex(DbContract.Speakers.GITHUB)),
-                    cur.getString(cur.getColumnIndex(DbContract.Speakers.LINKEDIN)),
-                    cur.getString(cur.getColumnIndex(DbContract.Speakers.ORGANISATION)),
-                    cur.getString(cur.getColumnIndex(DbContract.Speakers.POSITION)),
-                    cur.getString(cur.getColumnIndex(DbContract.Speakers.COUNTRY)),
-                    cur.getString(cur.getColumnIndex(DbContract.Speakers.SESSIONS))
-                    );
-            speakers.add(s);
+//            s = new Speaker(
```
Comment (leto) [inline app/src/main/java/org/fossasia/openevent/dbutils/DbSingleton.java]: Same thing, what is up here? You can put this work on a different branch if it is still in progress.

File: app/src/androidTest/java/org/fossasia/openevent/APITest.java (hunk 1 above)
Comment (mananwason) [inline app/src/androidTest/java/org/fossasia/openevent/APITest.java]: @leto It does pass. The error was in the first test, so I commented this but forgot to uncomment it after the error was fixed. I'll do it in my next commit.

File: app/src/main/java/org/fossasia/openevent/activities/MainActivity.java (hunk 2 above)
Comment (mananwason) [inline app/src/main/java/org/fossasia/openevent/activities/MainActivity.java]: I still have use for this. I did comment this as I used this in some other way.

File: app/src/main/java/org/fossasia/openevent/dbutils/DbSingleton.java (hunk 3 above)
Comment (mananwason) [inline app/src/main/java/org/fossasia/openevent/dbutils/DbSingleton.java]: @leto the thing that this commit was intended is the data retrieval from api and that is working perfectly. I have to make some changes in the database, so I commented this part until I get to it.

File: app/src/main/java/org/fossasia/openevent/dbutils/dataDownloadStart.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package org.fossasia.openevent.dbutils;
+
+import org.fossasia.openevent.api.APIClient;
+
+/**
+ * Created by MananWason on 26-05-2015.
+ */
+public class dataDownloadStart {
```
Comment (creativepsyco) [inline app/src/main/java/org/fossasia/openevent/dbutils/dataDownloadStart.java]: Class Names should begin with a capital letter. And `dataDownloadStart` has what purpose?

File: app/src/main/java/org/fossasia/openevent/dbutils/DbSingleton.java (hunk 3 above)
Comment (creativepsyco) [inline app/src/main/java/org/fossasia/openevent/dbutils/DbSingleton.java]: Don't commit commented code. Git is there for that purpose, if you go wrong you can always revert and recover your work. So don't be afraid to remove code that you comment.

File: app/src/main/java/org/fossasia/openevent/data/Event.java [hunk 5]
```diff
@@ -0,0 +1,7 @@
+package org.fossasia.openevent.data;
+
+/**
+ * Created by MananWason on 27-05-2015.
+ */
+public class Event {
```
Comment (creativepsyco) [inline app/src/main/java/org/fossasia/openevent/data/Event.java]: What's the purpose of this class?

File: app/src/main/java/org/fossasia/openevent/api/Urls.java [hunk 6]
```diff
@@ -8,11 +8,10 @@
...
-    public static final String BASE_URL = "http://championswimmer.in:8080";
+    public static final String BASE_URL = "http://glacial-earth-2730.herokuapp.com";
+    public static final String BASE_GET_URL = BASE_URL + "/get/api/" + API_VERSION;
 
     public static abstract class Get {
-        public static final String BASE_GET_URL = BASE_URL + "/get/api/" + API_VERSION;
-
         public static final String SESSIONS = BASE_GET_URL + "/sessions";
         public static final String SPONSORS = BASE_GET_URL + "/sponsors";
         public static final String SPEAKERS = BASE_GET_URL + "/speakers";
```
Comment (creativepsyco) [inline app/src/main/java/org/fossasia/openevent/api/Urls.java]: We can remove this.

File: app/src/main/java/org/fossasia/openevent/api/processor/EventListResponseProcessor.java [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+import retrofit.client.Response;
+
+/**
+ * Created by MananWason on 27-05-2015.
+ */
+public class EventListResponseProcessor implements Callback<EventResponseList> {
+    private static final String TAG = "Events";
+    @Override
+    public void success(EventResponseList eventResponseList, Response response) {
+        Log.d(TAG," done");
```
Comment (creativepsyco) [inline app/src/main/java/org/fossasia/openevent/api/processor/EventListResponseProcessor.java]: You can check out https://github.com/JakeWharton/timber for Log related stuff. `done` within the Log message means nothing, It should be used only for development.

File: app/src/main/java/org/fossasia/openevent/api/processor/MicrolocationListResponseProcessor.java [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+/**
+ * Created by MananWason on 27-05-2015.
+ */
+public class MicrolocationListResponseProcessor implements Callback<MicrolocationResponseList> {
+    private static final String TAG = "Microprocessor";
+
+    @Override
+    public void success(MicrolocationResponseList microlocationResponseList, Response response) {
+        Log.d(TAG," Done");
```
Comment (creativepsyco) [inline app/src/main/java/org/fossasia/openevent/api/processor/MicrolocationListResponseProcessor.java]: Same here.

File: app/src/main/java/org/fossasia/openevent/api/processor/SpeakerListResponseProcessor.java [hunk 9]
```diff
@@ -1,5 +1,8 @@
 package org.fossasia.openevent.api.processor;
 
+import android.nfc.Tag;
```
Comment (creativepsyco) [inline app/src/main/java/org/fossasia/openevent/api/processor/SpeakerListResponseProcessor.java]: We have a dependency on nfc package?

File: app/src/main/java/org/fossasia/openevent/data/Microlocation.java [hunk 10]
```diff
@@ -0,0 +1,62 @@
+package org.fossasia.openevent.data;
+
+/**
+ * Created by MananWason on 26-05-2015.
+ */
+public class Microlocation {
+    int id;
+    String name;
+    long latitude;
```
Comment (creativepsyco) [inline app/src/main/java/org/fossasia/openevent/data/Microlocation.java]: `latitude` and `longitude` are always `float`

**Comment by creativepsyco** [comment]
Hi,

@mananwason There is whole bunch of commented code thats needs to be removed. And issues which @leto have raised as well as some semantic ones like longitude is a `long` when it is mostly `float` everywhere.

Second, inconsistencies in class-names, importing of packages that you don't need, Log messages do not contain useful messages, tests are missing, classes are not complete in terms of functionality. etc. 

Also, coding styles are not consistent. You can download [my code styles][1] and import them in Android Studio using File > Import Settings. After that, go to Preferences > Editor > Code Style and enable the OSS code style. Before committing make sure to reformat code according to style.

Another major thing is that you have a merge commit in your Pull Request, this can only happen if the current fork's `master` branch was merged with yours. You shouldn't do that. I understand that you need to maintain the freshness of your fork but you may want to rebase your fork instead of merging, otherwise the merge commit is always going to end up in the Pull Request. Follow this [tutorial][2] to perform fork, branch, squash and file a PR.

At the end of the day, I would like to see a single commit that can be merged :smile: 

[1]: https://dl.dropboxusercontent.com/u/10123399/settings.jar
[2]: https://gun.io/blog/how-to-github-fork-branch-and-pull-request/

File: app/src/main/java/org/fossasia/openevent/data/Event.java (hunk 5 above)
Comment (mananwason) [inline app/src/main/java/org/fossasia/openevent/data/Event.java]: It was just to get events as an object and make lists of events

File: app/src/main/java/org/fossasia/openevent/api/processor/EventListResponseProcessor.java (hunk 7 above)
Comment (mananwason) [inline app/src/main/java/org/fossasia/openevent/api/processor/EventListResponseProcessor.java]: Yeah I did use it for development only, but forgot to remove it.

**Comment by mananwason** [comment]
Thanks @creativepsyco I'll make the changes you and @leto suggested and then I'll rebase and create a PR.

**Pull request closed without merging**